packages/legacy-scripting-runner/index.js

```diff
@@ -1389,15 +1389,17 @@ const legacyScriptingRunner = (Zap, zcli, input) => {
 
   const runHydrateFile = (bundle) => {
     const meta = bundle.inputData.meta || {};
-    const requestOptions = bundle.inputData.request || {};
 
-    // Legacy z.dehydrateFile(url, request, meta) behavior: if request argument is
-    // provided, the dev is responsible for doing auth themselves, so we use an internal
-    // request client to avoid running the app's http middlewares.
-    const request = _.isEmpty(requestOptions)
-      ? zcli.request
-      : createInternalRequestClient(input);
+    // Legacy z.dehydrateFile(url, request, meta) behavior: if request argument
+    // is provided, the dev is responsible for doing auth themselves, so we use
+    // an internal request client to avoid running the app's http middlewares.
+    const request =
+      bundle.inputData.request === null ||
```

Comment:
nit, but you can do `bundle.inputData.request == null` (double eq), which covers both `null` and `undefined`

Reply:
I'll merge this PR as-is because I don't want to wait for another review. But that's another "good" thing to know about JavaScript. Thanks!